idaes/core/util/model_diagnostics.py

```diff
@@ -3363,7 +3356,7 @@ def b_rule(b, i):
         val = value(inf_prob.y[i])
         if abs(val) < cutoff:
             break
-        ill_cond.append((i, val))
+        ill_cond.append((components[i].name, val))
```

Comment:
Seeing as we have the component here, I would suggest we return that rather than the name from this method. This gives us a bit more flexibility later on in working with the results, and any display method can easily get the name from the component.

Reply:
Good idea. Done in 5316a01.